src/types.ts

```diff
@@ -117,6 +121,27 @@ export interface IFrameApiRequest<Q> {
 export interface DefinitionWithKey {
     key: string;
 }
+
+export interface MenuItemCommon extends DefinitionWithKey {
+    label: string;
+}
+
+export interface LinkMenuItem extends MenuItemCommon {
+    type: MenuItemType.LINK;
+    href: string;
+}
+
+export interface EventMenuItem extends MenuItemCommon {
+    type: MenuItemType.EVENT;
+}
+
+export type MenuItem = LinkMenuItem | EventMenuItem;
+
+export interface MenuItemRequestResponse {
+    items: MenuItem[];
```

Comment:
I don't have a specific usecase in mind, I guess this was motivated by experience with API design (since this is an API of sorts). If your API returns an array and it turns out you need some other data about the array later, you'd have to make a breaking API change. If it's an object, and you need some new piece of data you just add a key